### PR TITLE
Fix alpha channel in gifit

### PR DESCRIPTION
**Pull request by davidschlegel**
If the plot contains non-trivial alpha values (e.g. by setting an transparent figure or axis background using `matplotlib.rcParams`), now this is reflected in `gifit`.

Before, all alpha values were discarded.

### CODE REVIEW

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
Thanks a lot! Unfortunately this re-introduces an old color quantization bug (see https://github.com/dynamiqs/dynamiqs/pull/723). If you find a way around we'll be happy to merge it.

You can test with
```python
import dynamiqs as dq
import jax.numpy as jnp

n = 16
a = dq.destroy(n)
H = dq.zeros(n)
jump_ops = [a @ a - 4.0 * dq.eye(n)]  # cat state inflation
psi0 = dq.coherent(n, 0)
tsave = jnp.linspace(0, 1.0, 1001)
result = dq.mesolve(H, jump_ops, psi0, tsave)
dq.plot.wigner_gif(result.states, fps=25, xmax=4.0, ymax=2.0)
```

Result on `main`:
34b40ef2-e34e-4057-b413-52cc9f56ee47

Result for your PR:
1ec1e578-358c-4623-b1f4-a215a478d5f3

**Comment by davidschlegel** [comment]
Okay, I have to see what actually causes this (the `RGBA` attribute or the `disposal` argument)

**Comment by davidschlegel** [comment]
Actually, I see the issue right [here](https://github.com/python-pillow/Pillow/blob/2755e0ffaadc8b29c3e67e223c333c50e197a733/src/PIL/Image.py#L964-L965):
```python
if mode == "P" and self.mode == "RGBA":
      return self.quantize(colors)
```

I am trying to find a workaround.

**Comment by davidschlegel** [comment]
It seems a little more tricky.
Quantizing RGBA values can be done via `libimagequant`, but this is not natively included in Pillow, as it has a different licence. For it to work requires building Pillow from source against libimagequant.
Is this something that we could do?

**Comment by pierreguilmin** [comment]
> It seems a little more tricky. Quantizing RGBA values can be done via `libimagequant`, but this is not natively included in Pillow, as it has a different licence. For it to work requires building Pillow from source against libimagequant. Is this something that we could do?

This would be a bit too advanced for just this specific feature, especially since we'd need to include it in the CI and the installation process. So no this is not something we can do.

**Comment by pierreguilmin** [comment]
Maybe @gautierronan or @abocquet have ideas on how to fix this?
